iron-collapse.html

```diff
@@ -142,17 +151,39 @@
     },
 
     updateSize: function(size, animated) {
-      this.enableTransition(animated);
-      var s = this.style;
-      var nochange = s[this.dimension] === size;
-      s[this.dimension] = size;
-      if (animated && (nochange || !this._enableTransition)) {
-        this._transitionEnd();
+      // No change!
+      if (this.style[this.dimension] === size) {
+        return;
+      }
+
+      this._updateTransition(false);
+      // If we can animate, must do some prep work.
+      if (animated && !this.noAnimation) {
+        // Animation will start at the current size.
+        var startSize = this._calcSize();
+        // For `auto` we must calculate what is the final size for the animation.
+        // After the transition is done, _transitionEnd will set the size back to `auto`.
+        if (size === 'auto') {
+          this.style[this.dimension] = size;
+          size = this._calcSize();
+        }
+        // Go to startSize without animation.
+        this.style[this.dimension] = startSize;
+        // Force layout to ensure transition will go.
+        /** @suppress {suspiciousCode} */ this.offsetHeight;
+        // Enable animation.
+        this._updateTransition(true);
       }
+      // Set the final size.
+      this.style[this.dimension] = size;
     },
 
     enableTransition: function(enabled) {
-      this.style.transitionDuration = (enabled && this._enableTransition) ? '' : '0s';
+      this.noAnimation = !enabled;
```

Comment:
I'd add a comment here maybe that the only reason this function exists is to maintain backwards compatibility, and maybe they should just set `noAnimation` directly instead.